### PR TITLE
Watchdog API redesign and shim for mcux wdog driver

### PR DESCRIPTION
**Pull request by MaureenHelm**
Builds upon the work in #7007 and #7163 to add an mcux watchdog shim driver. The watchdog test passes on frdm_k64f, hexiwear_k64, and usb_kw24d512 boards.

cc: @kl-cruz

### CODE REVIEW

**Comment by MaureenHelm** [comment]
Rebased to @kl-cruz's latest

**Review by dbkinder — APPROVED** [review]
It looks like the doc changes are OK.

**Comment by MaureenHelm** [comment]
Rebased to master and removed DNM/RFC labels now that #7007 is merged.

@anangl @kl-cruz Will you please review?

File: drivers/watchdog/wdt_mcux_wdog.c [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+ * Copyright (c) 2018, NXP
+
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <watchdog.h>
+#include <clock_control.h>
+#include <fsl_wdog.h>
+
+#define SYS_LOG_LEVEL SYS_LOG_WDT_LEVEL
```
Comment (anangl) [inline drivers/watchdog/wdt_mcux_wdog.c]: Shouldn't it be `CONFIG_SYS_LOG_WDT_LEVEL`?

**Comment by anangl** [comment]
Apart from the one doubt mentioned above, LGTM.

File: drivers/watchdog/wdt_mcux_wdog.c (hunk 1 above)
Comment (MaureenHelm) [reply drivers/watchdog/wdt_mcux_wdog.c]: Good catch, thanks. Fixed.

**Review by anangl — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by kl-cruz — COMMENTED** [review]
Looks good to me

**Comment by MaureenHelm** [comment]
Rebased to latest master and fixed merge conflicts

**Pull request merged**